database/database_handler.py

```diff
@@ -111,3 +93,48 @@ def managed_session(self):
             raise e
         finally:
             session.close()
+
+    def close_connection(self):
+        try:
+            # Dispose the engine
+            self.engine.dispose()
```

Comment:
Do we need to manually dispose the engine?

Reply:
The pipeline failed when I used close_connection

The "with statement" in val.py (with db_config.managed_session() as session) ensures that the session is properly closed and any resources associated with it are released when you exit the block. SQLAlchemy takes care of closing the session, so we don't need to manually call close or dispose.

Remove this function?